libraries/bezier-rs/src/svg.rs

```diff
@@ -0,0 +1,74 @@
+/// Structure to represent optional parameters that can be passed to the `into_svg` function.
+pub struct ToSVGOptions {
+	/// Color of the line segments along the `Subpath`. Defaulted to `black`.
+	pub curve_stroke_color: String,
+	/// Width of the line segments along the `Subpath`. Defaulted to `2.`.
+	pub curve_stroke_width: f64,
+	/// Stroke color outlining circles marking anchors on the `Subpath`. Defaulted to `black`.
+	pub anchor_stroke_color: String,
+	/// Stroke width outlining circles marking anchors on the `Subpath`. Defaulted to `2.`.
+	pub anchor_stroke_width: f64,
+	/// Radius of the circles marking anchors on the `Subpath`. Defaulted to `4.`.
+	pub anchor_radius: f64,
+	/// Fill color of the circles marking anchors on the `Subpath`. Defaulted to `white`.
+	pub anchor_fill: String,
+	/// Color of the line segments connecting anchors to handle points. Defaulted to `gray`.
+	pub handle_line_stroke_color: String,
```

Comment:
Could we make some of these `String`s `&'static str`s to reduce heap allocations? I don't really see a case where these should be generated at runtime.

Reply:
I can't say for other people's use cases for the `to_svg` function, but we're currently trying to replace all our HTML canvas usages to this svg handler. For the most part, we're mainly using colours like red, black, gray, etc. But there's a few `bezier-rs` demos that use something akin to `hsl(${40 * idx}, 100%, 50%)` to create a rainbow effect with variable number of colours. 

Do you have any recommendations in mind for how to replicate this if we change the `String`s to `&'static str`s?

Reply:
Actually perhaps if you provide a public function to write just the string inside the `d="..."` attribute then this means that graphite can apply whatever styles we like (e.g. a dash array) without increasing the scope of the library beyond bezier curves. Then this struct isn't so performance critical.

Reply:
I like this plan best.

Reply:
In this case, would it be better to split the `to_svg` function up into different ones that draw:

1. The curve itself
2. The curve's endpoints
4. The curve's handles
4. The lines from the endpoints to the handles

In this way, each function will have only one string parameter to write (i.e. the `"d=..."` attribute of a specific path/shape/HTML entity, assuming that endpoints and such should have the same styling). Additionally, the users have the flexibility to choose which of those features they want drawn in the SVG without needing to overcomplicate function parameters/options.

Reply:
Sounds good.

Reply:
I've changed the `Bezier` code to no longer use this struct. The `Subpath` logic still uses `ToSVGOption` at the moment, but I'll work on that after this refactor is finalized.